### PR TITLE
add via keymap for boardsource/3x4

### PR DESCRIPTION
**Pull request by gwillad**
## Add boardsource/3x4 keymap for via support

In /keyboards/boardsource/3x4/keymaps added

- via/
- via/keymap.c which contains the same layout as keymaps/default/keymaps.c
- via/rules.mk which enables via
- via/readme.md

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* changed boardsource/3x4 product id to be unique

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/boardsource/3x4/rules.mk [hunk 1]
```diff
@@ -29,3 +29,6 @@ UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+LEADER_ENABLE = no
+KEY_LOCK_ENABLE = no
+TERMINAL_ENABLE = no
```
Comment (fauxpark) [inline keyboards/boardsource/3x4/rules.mk:34]:
```suggestion
```

File: keyboards/boardsource/3x4/rules.mk [hunk 2]
```diff
@@ -29,3 +29,6 @@ UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/boardsource/3x4/rules.mk:31]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/boardsource/3x4/rules.mk [hunk 3]
```diff
@@ -26,6 +26,5 @@ NKRO_ENABLE = no            # USB Nkey Rollover
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 MIDI_ENABLE = no            # MIDI support
 UNICODE_ENABLE = no         # Unicode
```
Comment (fauxpark) [inline keyboards/boardsource/3x4/rules.mk:28]:
```suggestion
```

File: keyboards/boardsource/3x4/config.h [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 /* USB Device descriptor parameter */
 #define VENDOR_ID       0xF7E0
```
Comment (noroadsleft) [inline keyboards/boardsource/3x4/config.h:8]:
```suggestion
#define VENDOR_ID       0x4273 // "Bs" - Boardsource
```

File: keyboards/boardsource/3x4/config.h [hunk 5]
```diff
@@ -6,7 +6,7 @@
...
-#define PRODUCT_ID      0x0412
+#define PRODUCT_ID      0x0413
```
Comment (noroadsleft) [inline keyboards/boardsource/3x4/config.h:9]: Based on `3x4`, obviously.

```suggestion
#define PRODUCT_ID      0x0304
```

File: keyboards/boardsource/3x4/config.h (hunk 4 above)
Comment (gwillad) [reply keyboards/boardsource/3x4/config.h:8]: should I take the time to change this for the other bourdsource boards - others aren't currently supported by VIA, so I'm assuming this wouldn't be a breaking change?

File: keyboards/boardsource/3x4/config.h (hunk 5 above)
Comment (gwillad) [reply keyboards/boardsource/3x4/config.h:9]: happy to make this change - are product id's unique across the project? Or just across a vendor?

**Comment by noroadsleft** [comment]
@gwillad,

> should I take the time to change this for the other boardsource boards - others aren't currently supported by VIA, so I'm assuming this wouldn't be a breaking change?

For the other Boardsource boards, you can if you want. If not I'll try to get it in the next couple of days.

> happy to make this change - are product id's unique across the project? Or just across a vendor?

The Vendor/Product pairings should be unique across the project (unless it's different revisions of the same keyboard), but this is something that we haven't kept track of over the years. Ideally each keyboard by a specific designer or vendor should share a Vendor ID, each with a unique Product ID.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**